_rules/inline-link-in-paragraph-has-distinguishable-border-36f116.md

```diff
@@ -0,0 +1,201 @@
+---
+id: 36f116
+name: Inline link in paragraph has distinguishable border
+rule_type: atomic
+
+description: |
+  This rule checks that links that are embedded in a paragraph, list or cell have a border that distinguishes them from the surrounding text
+
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+
+acknowledgements:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+  previous authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any HTML element which:
+
+- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- is a [descendant][] in the [flat tree][] of a `p`, `li` or `td` HTML elements or elements with a [semantic role][] of ['listitem'][listitem] or ['cell'][cell] (with the closest [ancestor][] in the [flat tree][] being henceforth referred to as the _ancestor_ element); and
+- has [descendant][] [text nodes][text node] in the [flat tree][] that are [visible][]; and
+- the _ancestor_ element has other [descendant][] [visible][] [text nodes][text node] in the [flat tree][].
+
+## Expectation
+
+Each target element has at least one of the `border-top-color`, `border-right-color`, `border-bottom-color` and `border-left-color` style properties with [width](https://drafts.csswg.org/css-backgrounds/#border-width) bigger than 0px and with a computed [color value](https://drafts.csswg.org/css-backgrounds/#border-color) different from the `background-color` of the target element.
```

Comment:
Neither of the properties `border-top-color`, `border-right-color`, `border-bottom-color` and `border-left-color` are widths; they're all colors. We therefore cannot say "with [width](https://drafts.csswg.org/css-backgrounds/#border-width) bigger than 0px".

Reply:
What is `background-color` referring to here? Also make sure that we're comparing apples to apples; are we looking at specified, computed, or used values?

Reply:
Nice catch